src/main/java/org/broadinstitute/consent/http/service/TDRService.java

```diff
@@ -74,9 +73,15 @@ public ApprovedUsers getApprovedUsersForDataset(AuthUser authUser, Dataset datas
         .filter(email -> !email.isBlank())
         .toList();
 
-    List<ApprovedUser> approvedUsers = Stream.of(labCollaborators, userEmails)
+    // Filter to users where that have a library card
+    List<String> allEmails = Stream.of(labCollaborators, userEmails)
         .flatMap(List::stream)
         .distinct()
+        .toList();
+
+    List<ApprovedUser> approvedUsers = libraryCardDAO.findByUserEmails(allEmails)
+        .stream()
+        .map(LibraryCard::getUserEmail)
```

Comment:
Looks good.  

Took me a few to walk the path of the library card email address, and I understand you're using that address because we have library cards issued to non-users who might be granted access.

The thing I'm left thinking about is if we should be checking the library card email address to see if it's got a valid institution in the institution map.  For non-users with library cards, we'll never perform revalidation of the library card to institutional mapping because those individuals never likely log into DUOS.